README.md
# ChangeBlog

My blog for documenting interesting things and personal projects.

The blog  was previously a [Wordpress](http://wordpress.com/) site running on [Dreamhost](http://dreamhost.com/) servers. I eventually moved to [Jekyll](https://github.com/mojombo/jekyll) hosted on [Github](http://github.com/) servers with images served by [Flickr](http://flickr.com/).

Moving away from Wordpress gave the site a noticable performance boost at no cost. I highly recommend this solution to any developer interested in running a blog.

## Theme Installation

Installation of the hugo theme must be done separately.  Issue the following commands after cloning the repo.

```bash
cd themes
git clone https://github.com/MarcusVirg/forty
```

## Support

[![ko-fi](https://ko-fi.com/img/githubbutton_sm.svg)](https://ko-fi.com/V7V5F5X7E)



CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

## [0.1.10] - 2022-11-10
### Added
- Add daily flash video posts for 2021 (Oct, Nov, Dec) and 2022 (Jan, Sept, Oct)
- Add RSS field for project link
- Add new rss.xml layout so subtitle is included in title

### Changed
- Update Storage Case Base Template (SCBT) post
- Replace description param with summary param
- Update README

## [0.1.9] - 2022-09-23
### Added
- Add Storage Case Base Template (SCBT) post
- Add daily flash video posts for Feb, Mar, Apr
- Add ko-fi donation links
- Add YouTube social link to footer

### Changed
- Update README

## [0.1.8] - 2022-09-02
### Added
- Add daily flash video posts for May, Jun, Jul, Aug
- Add related tags links for posts
- Add photos category

### Changed
- Update .jpg images to .webp
- Update footer links
- Change css files to load asynchronously

## [0.1.7] - 2021-12-02
### Added
- Author name added to meta description
- Pinterest and Thingiverse  social link added to footer
- Created new post for Daily Fla.sh (1 Second Everyday inspired) tool
- List view for categories and taxonomies added 
- Projects link added to collapsible menu
- Create partial for displaying post information

## [0.1.6] - 2021-10-15
### Added
- AngelList and Bitbucket social link added to footer
- Created new post for Verito Google Photos validation tool

### Changed
- Update content in solder fume extractor post
- Updated banner image

## [0.1.5] - 2021-07-31
### Added
- Automated creation of small, medium and large image sizes using `<srcset>` tag

### Changed
- Update existing content images to use `<srcset>` shortcode tag

## [0.1.4] - 2021-07-17
### Added
- New post (DIY Solder Fume Extractor) published
- Add ability to use emojis in posts
- Allow rendering of RAW HTML so we can use HTML tags like `<img>`, etc

### Changed
- Update content on main, about and securing changeblog images posts 

## [0.1.3] - 2021-06-27
### Added
- Offer up alansolidum.com as canonical link
- Add front matter subtitle to be part of title tag
- Add post customizable meta-description tag

### Changed
- Only allow GoogleOpt/TagManager in production 

## [0.1.2] - 2021-06-11
### Added
- About page with developer info added
- Functionality for Google Optimize and Tag Manager added

### Changed
- Nav bar disappears on user scrolling down and reappears when user scrolls up
- Previous about page showing tech used on site renamed to about-site

## [0.1.1] - 2021-06-04
### Added
- About page displaying old site info and current site technology

### Fixed
- SSL active and baseURL uses `https`
- All footer social links now have correct URLs

### Removed
- Dead links removed/disabled

## [0.1.0] - 2021-05-29
### Added
- Site (using Hugo with forty theme) deployed on Gitlab pages
- Two posts (Securing Changeblog Images and Making Ornaments) published
- Contact form enabled


static/js/main.js
(function($) {

	skel.breakpoints({
		xlarge: '(max-width: 1680px)',
		large: '(max-width: 1280px)',
		medium: '(max-width: 980px)',
		small: '(max-width: 736px)',
		xsmall: '(max-width: 480px)',
		xxsmall: '(max-width: 360px)'
	});

	/**
	 * Applies parallax scrolling to an element's background image.
	 * @return {jQuery} jQuery object.
	 */
	$.fn._parallax = (skel.vars.browser == 'ie' || skel.vars.browser == 'edge' || skel.vars.mobile) ? function() { return $(this) } : function(intensity) {

		var	$window = $(window),
			$this = $(this);

		if (this.length == 0 || intensity === 0)
			return $this;

		if (this.length > 1) {

			for (var i=0; i < this.length; i++)
				$(this[i])._parallax(intensity);

			return $this;

		}

		if (!intensity)
			intensity = 0.25;

		$this.each(function() {

			var $t = $(this),
				on, off;

			on = function() {

				$t.css('background-position', 'center 100%, center 100%, center 0px');

				$window
					.on('scroll._parallax', function() {

						var pos = parseInt($window.scrollTop()) - parseInt($t.position().top);

						$t.css('background-position', 'center ' + (pos * (-1 * intensity)) + 'px');

					});

			};

			off = function() {

				$t
					.css('background-position', '');

				$window
					.off('scroll._parallax');

			};

			skel.on('change', function() {

				if (skel.breakpoint('medium').active)
					(off)();
				else
					(on)();

			});

		});

		$window
			.off('load._parallax resize._parallax')
			.on('load._parallax resize._parallax', function() {
				$window.trigger('scroll');
			});

		return $(this);

	};

	$(function() {

		var	$window = $(window),
			$body = $('body'),
			$wrapper = $('#wrapper'),
			$header = $('#header'),
			$banner = $('#banner');

		var prevScrollpos = window.pageYOffset;

		// Disable animations/transitions until the page has loaded.
			$body.addClass('is-loading');

			$window.on('load pageshow', function() {
				window.setTimeout(function() {
					$body.removeClass('is-loading');
				}, 100);
			});

		// Clear transitioning state on unload/hide.
			$window.on('unload pagehide', function() {
				window.setTimeout(function() {
					$('.is-transitioning').removeClass('is-transitioning');
				}, 250);
			});

		// Fix: Enable IE-only tweaks.
			if (skel.vars.browser == 'ie' || skel.vars.browser == 'edge')
				$body.addClass('is-ie');

		// Fix: Placeholder polyfill.
			$('form').placeholder();

		// Prioritize "important" elements on medium.
			skel.on('+medium -medium', function() {
				$.prioritize(
					'.important\\28 medium\\29',
					skel.breakpoint('medium').active
				);
			});

		// Scrolly.
			$('.scrolly').scrolly({
				offset: function() {
					return $header.height() - 2;
				}
			});

		// Tiles.
			var $tiles = $('.tiles > article');

			$tiles.each(function() {

				var $this = $(this),
					$image = $this.find('.image'), $img = $image.find('img'),
					$link = $this.find('.link'),
					x;

				// Image.

					// Set image.
						$this.css('background-image', 'url(' + $img.attr('src') + ')');

					// Set position.
						if (x = $img.data('position'))
							$image.css('background-position', x);

					// Hide original.
						$image.hide();

				// Link.
					if ($link.length > 0) {

						$x = $link.clone()
							.text('')
							.addClass('primary')
							.appendTo($this);

						$link = $link.add($x);

						$link.on('click', function(event) {

							var href = $link.attr('href');

							// Prevent default.
								event.stopPropagation();
								event.preventDefault();

							// Start transitioning.
								$this.addClass('is-transitioning');
								$wrapper.addClass('is-transitioning');

							// Redirect.
								window.setTimeout(function() {

									if ($link.attr('target') == '_blank')
										window.open(href);
									else
										location.href = href;

								}, 500);

						});

					}

			});

            $window.on('scroll', function() {
				var currentScrollPos = window.pageYOffset;
                if (prevScrollpos < currentScrollPos) {
                    $header.addClass('alt'); 
                } else {
				    $header.removeClass('alt'); $header.addClass('reveal');
                }
                prevScrollpos = currentScrollPos;
			});


		// Header.
			if (skel.vars.IEVersion < 9)
				$header.removeClass('alt');

		// Banner.
			$banner.each(function() {

				var $this = $(this),
					$image = $this.find('.image'), $img = $image.find('img');

				// Parallax.
					$this._parallax(0.275);

				// Image.
					if ($image.length > 0) {

						// Set image.
							$this.css('background-image', 'url(' + $img.attr('src') + ')');

						// Hide original.
							$image.hide();

					}

			});

		// Menu.
			var $menu = $('#menu'),
				$menuInner;

			$menu.wrapInner('<div class="inner"></div>');
			$menuInner = $menu.children('.inner');
			$menu._locked = false;

			$menu._lock = function() {

				if ($menu._locked)
					return false;

				$menu._locked = true;

				window.setTimeout(function() {
					$menu._locked = false;
				}, 350);

				return true;

			};

			$menu._show = function() {

				if ($menu._lock())
					$body.addClass('is-menu-visible');

			};

			$menu._hide = function() {

				if ($menu._lock())
					$body.removeClass('is-menu-visible');

			};

			$menu._toggle = function() {

				if ($menu._lock())
					$body.toggleClass('is-menu-visible');

			};

			$menuInner
				.on('click', function(event) {
					event.stopPropagation();
				})
				.on('click', 'a', function(event) {

					var href = $(this).attr('href');

					event.preventDefault();
					event.stopPropagation();

					// Hide.
						$menu._hide();

					// Redirect.
						window.setTimeout(function() {
							window.location.href = href;
						}, 250);

				});

			$menu
				.appendTo($body)
				.on('click', function(event) {

					event.stopPropagation();
					event.preventDefault();

					$body.removeClass('is-menu-visible');

				})
				.append('<a class="close" href="#menu">Close</a>');

			$body
				.on('click', 'a[href="#menu"]', function(event) {

					event.stopPropagation();
					event.preventDefault();

					// Toggle.
						$menu._toggle();

				})
				.on('click', function(event) {

					// Hide.
						$menu._hide();

				})
				.on('keydown', function(event) {

					// Hide on escape.
						if (event.keyCode == 27)
							$menu._hide();

				});

	});

})(jQuery);